stock_measuring_device_zippcube/controllers/main.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+import logging
+import os
+
+from odoo import _, http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class ZippcubeController(http.Controller):
+    DEVICE_SECRET_KEY = "ZIPPCUBE_SECRET"
+
+    weight_keys = ("weight",)
+    measures_keys = ("length", "width", "height")
+    expected_keys = ("secret", "barcode") + weight_keys + measures_keys
+
+    @http.route(
+        "/stock/zippcube/<string:zippcube_device_name>/measurement",
+        type="json",
+        auth="none",
+    )
+    def measurement(self, zippcube_device_name):
+        data = request.jsonrequest
+        _logger.info("/measurement, data received: {}".format(data))
+
+        env = request.env(su=True)
+        device = env["measuring.device"].search(
+            [("name", "=", zippcube_device_name), ("device_type", "=", "zippcube")],
+            limit=1,
+        )
+        if not device:
+            raise MissingError(
+                _("No such Zippcube with name {}.").format(zippcube_device_name)
+            )
+
+        keys_missing = set(self.expected_keys) - set(data)
```

Comment:
A validator could save some coding/time here  eg: https://docs.python-cerberus.org/en/stable/schemas.html
Not a change request, of course :wink: